Classes/Flowpack/ElasticSearch/ContentRepositoryAdaptor/Eel/ElasticSearchQueryBuilder.php

```diff
@@ -80,71 +83,24 @@ class ElasticSearchQueryBuilder implements QueryBuilderInterface, ProtectedConte
 
     /**
      * The ElasticSearch request, as it is being built up.
-     *
-     * @var array
+     * @var QueryInterface
      */
-    protected $request = [
-        // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-query.html
-        'query' => [
-            // The top-level query we're working on is a *filtered* query, as this allows us to efficiently
-            // apply *global constraints* in the form of *filters* which apply on the whole query.
-            //
-            // NOTE: we do NOT add a search request FILTER to the query currently, because that would mean
-            // that the filters ONLY apply for query results, but NOT for facet calculation (as explained on
-            // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/search-request-filter.html)
-            //
-            // Reference: http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-filtered-query.html
-            'filtered' => [
-                'query' => [
-                    'bool' => [
-                        'must' => [
-                            [
-                                'match_all' => []
-                            ]
-                        ]
-                    ]
-
-                ],
-                'filter' => [
-                    // http://www.elasticsearch.org/guide/en/elasticsearch/reference/current/query-dsl-bool-filter.html
-                    'bool' => [
-                        'must' => [],
-                        'should' => [],
-                        'must_not' => [
-                            // Filter out all hidden elements
-                            [
-                                'term' => ['_hidden' => true]
-                            ],
-                            // if now < hiddenBeforeDateTime: HIDE
-                            // -> hiddenBeforeDateTime > now
-                            [
-                                'range' => [
-                                    '_hiddenBeforeDateTime' => [
-                                        'gt' => 'now'
-                                    ]
-                                ]
-                            ],
-                            [
-                                'range' => [
-                                    '_hiddenAfterDateTime' => [
-                                        'lt' => 'now'
-                                    ]
-                                ]
-                            ],
-                        ],
-                    ]
-                ]
-            ]
-        ],
-        'fields' => ['__path']
-    ];
+    protected $request;
 
     /**
      * @var array
      */
     protected $result = [];
 
     /**
+     * Initialize Object
+     */
+    public function initializeObject()
+    {
+        $this->request = $this->objectManager->get(QueryInterface::class, $this);
```

Comment:
There are multiple implementations of that interface, how should the ObjectManager know which one to pick? The functional tests all fail, saying `TYPO3\Flow\Object\Exception\UnknownObjectException: Object "Flowpack\ElasticSearch\ContentRepositoryAdaptor\Domain\Model\QueryInterface" is not registered.`